gap/base/methods.gi

```diff
@@ -0,0 +1,59 @@
+#############################################################################
+##
+##  This file is part of recog, a package for the GAP computer algebra system
+##  which provides a collection of methods for the constructive recognition
+##  of groups.
+##
+##  This files's authors include Sergio Siccha.
+##
+##  Copyright of recog belongs to its developers whose names are too numerous
+##  to list here. Please refer to the COPYRIGHT file for details.
+##
+##  SPDX-License-Identifier: GPL-3.0-or-later
+##
+##
+##  Implementation of recog methods
+##
+#############################################################################
+
+# HACK so that we can still treat RecogMethod objects as records
+InstallMethod(\., [IsRecogMethod, IsObject],
+function(x, nr)
+    local name, map;
+    name := NameRNam(nr);
+    map :=  rec(stamp := "Stamp",
+                method := "FunctionOfRecogMethod",
+                comment := "Comment");
+    if name in RecNames(map) then
+        return x!.(map.(name));
+    fi;
+    return x!.(name);
+end);
+
+# TODO: DisplayString, PrintString
+InstallMethod(ViewString, [IsRecogMethod],
+function(m)
+    local s;
+    s := "<RecogMethod \"";
+    Append(s, Stamp(m));
+    Append(s, "\": ");
+    Append(s, ViewString(FunctionOfRecogMethod(m)));
+    Append(s, ">");
+    return s;
+end);
+
+InstallGlobalFunction(RecogMethod,
+function(stamp, comment, func)
+    local r;
+    r := rec();
+    Objectify(RecogMethodType, r);
+    SetStamp(r, stamp);
+    SetComment(r, comment);
+    SetFunctionOfRecogMethod(r, func);
+    return r;
+end);
+
+InstallGlobalFunction(CallRecogMethod,
+function(m, args)
+    return CallFuncList(FunctionOfRecogMethod(m), args);
```

Comment:
If you change this to do something like `if IsRecord(m) then ... elif IsRecogMethod(m) then ... else Error() fi;` then you could immediately convert code to use it, right?

Alternatively, this could become an operation with several methods. Doesn't really matter, it's only temporary.